ui/appui-react/src/appui-react/statusfields/SnapMode.tsx

```diff
@@ -29,50 +29,63 @@ import { StatusBarPopover } from "../statusbar/popup/StatusBarPopover";
 interface SnapModeFieldEntry {
   label: string;
   value: number;
-  iconName: string;
+  iconSpec: string;
 }
 
-// eslint-disable-next-line deprecation/deprecation
-function getIconFromIconName(iconName: string): IconSpec {
-  let iconSpec = snapModeKeypoint;
-  switch (iconName) {
-    case "snaps":
-      iconSpec = snapModeKeypoint;
-      break;
-    case "snaps-intersection":
-      iconSpec = snapModeIntersection;
-      break;
-    case "snaps-center":
-      iconSpec = snapModeCenter;
-      break;
-    case "snaps-nearest":
-      iconSpec = snapModeNearest;
-      break;
-    case "snaps-origin":
-      iconSpec = snapModeOrigin;
-      break;
-    case "snaps-midpoint":
-      iconSpec = snapModeMidpoint;
-      break;
-    case "snaps-bisector":
-      iconSpec = snapModeBisector;
-      break;
-  }
-  return iconSpec;
-}
+// Field entry of all possible snap modes.
+const allSnapModeFieldEntries: SnapModeFieldEntry[] = [
+  {
+    label: "snapModeField.keypoint",
+    value: SnapMode.NearestKeypoint as number,
+    iconSpec: snapModeKeypoint,
+  },
+  {
+    label: "snapModeField.intersection",
+    value: SnapMode.Intersection as number,
+    iconSpec: snapModeIntersection,
+  },
+  {
+    label: "snapModeField.center",
+    value: SnapMode.Center as number,
+    iconSpec: snapModeCenter,
+  },
+  {
+    label: "snapModeField.nearest",
+    value: SnapMode.Nearest as number,
+    iconSpec: snapModeNearest,
+  },
+  {
+    label: "snapModeField.origin",
+    value: SnapMode.Origin as number,
+    iconSpec: snapModeOrigin,
+  },
+  {
+    label: "snapModeField.midpoint",
+    value: SnapMode.MidPoint as number,
+    iconSpec: snapModeMidpoint,
+  },
+  {
+    label: "snapModeField.bisector",
+    value: SnapMode.Bisector as number,
+    iconSpec: snapModeBisector,
+  },
+];
 
-/** Return icon class name for a specific snapMode. */
-const getSnapModeIconNameFromMode = (
-  modeEntries: SnapModeFieldEntry[],
-  snapMode: number
-) => {
-  for (const modeEntry of modeEntries) {
-    if (modeEntry.value === snapMode) return modeEntry.iconName;
-  }
-
-  if (snapMode > 0) return "snaps-multione";
+/** Return icon for a specific snapMode. */
+const getSnapModeIcon = (snapMode: number) => {
+  // Get all the modes present in the bitmask.
+  const modes: SnapModeFieldEntry[] = allSnapModeFieldEntries.filter(
+    (entry) => {
+      return (entry.value & snapMode) === entry.value;
+    }
+  );
 
-  return "placeholder";
+  // eslint-disable-next-line deprecation/deprecation
+  return modes.length === 1 ? (
+    <Icon iconSpec={modes[0].iconSpec} />
+  ) : (
+    <Icon iconSpec={snapModeKeypoint} />
+  );
```

Comment:
Is this needed for this PR?

Reply:
With the refactor I needed to rework the function to get the icon. 
I accounted for what will happen when multiple snap modes are active at the same time.

Reply:
I understand that but this PR does not add multiple active snap modes. It's just a nitpick that this code would fit better in the next PR but it is not a big deal. Leaving the choice up to you, not  a big deal for us.

Reply:
It was something to always take into account because the consumer can set multiple snapmode via props. There was already a for multiple snap mode. 

The next PR, for the next major, will only change the default behaviour to allow multiple snaps.